packages/store/src/stores/store-utils.js

```diff
@@ -106,3 +106,24 @@ export const provide = (mapStore, key, makeValue) => {
   return mapStore.get(key);
 };
 harden(provide);
+
+/**
+ * Call `provideAsync` to get or make the value associated with the key,
+ * when the maker is asynchronous.
+ * If there already is one, return that. Otherwise,
+ * call `makeValue(key)`, remember it as the value for
+ * that key, and return it.
+ *
+ * @template K,V
+ * @param {MapStore<K,V>} mapStore
+ * @param {K} key
+ * @param {(key: K) => Promise<V>} makeValue
+ * @returns {Promise<V>}
+ */
+export const provideAsync = async (mapStore, key, makeValue) => {
+  if (!mapStore.has(key)) {
+    mapStore.init(key, await makeValue(key));
```

Comment:
this goes against [No Nested Await](https://github.com/Agoric/agoric-sdk/wiki/No-Nested-Await). Is there any particular justification?

Reply:
The introduction of the `pending` Map doesn't seem equivalent to this. If it's essential, that seems to mean the nested await form was buggy. Anyway... it looks workable, and I'm content that #5728 is where we deal with it definitively.